server/rest/src/test/java/org/infinispan/rest/search/SearchCountClusteredTest.java

```diff
@@ -0,0 +1,366 @@
+      assertTotalAndPageSize(result, NOT_INDEXED_ENTRIES, 5L);
+   }
+
+   @Test
+   public void testMaxResultsNotIndexed() {
+      CompletionStage<RestResponse> result = queryWithoutPagination(nonIndexedCache(), "FROM NotIndexedEntity");
+      assertTotalAndPageSize(result, NOT_INDEXED_ENTRIES, NOT_INDEXED_ENTRIES);
+   }
+
+   @Test
+   public void testSortedNotIndexed() {
+      CompletionStage<RestResponse> result = queryWithoutPagination(nonIndexedCache(), "FROM NotIndexedEntity ORDER BY field2");
+      assertTotalAndPageSize(result, NOT_INDEXED_ENTRIES, NOT_INDEXED_ENTRIES);
+   }
+
+   @Test
+   public void testSortedNotIndexedPaginated() {
+      CompletionStage<RestResponse> result = queryWithDefaultPagination(nonIndexedCache(), "FROM NotIndexedEntity ORDER BY field2");
+      assertTotalAndPageSize(result, NOT_INDEXED_ENTRIES, DEFAULT_PAGE_SIZE);
+   }
+
+   @Test
+   public void testPaginatedNotIndexed() {
+      CompletionStage<RestResponse> result = queryWithPagination(nonIndexedCache(), "SELECT field1 FROM NotIndexedEntity", 5, 2);
+      assertTotalAndPageSize(result, NOT_INDEXED_ENTRIES, 5L);
+   }
+
+   @Test
+   public void testSelectNotIndexed() {
+      CompletionStage<RestResponse> result = queryWithDefaultPagination(nonIndexedCache(), "SELECT field1 FROM NotIndexedEntity");
+      assertTotalAndPageSize(result, NOT_INDEXED_ENTRIES, DEFAULT_PAGE_SIZE);
+   }
+
+   @Test
+   public void testSelectNotIndexedPaginated() {
+      CompletionStage<RestResponse> result = queryWithDefaultPagination(nonIndexedCache(), "SELECT field1 FROM NotIndexedEntity");
+      assertTotalAndPageSize(result, NOT_INDEXED_ENTRIES, DEFAULT_PAGE_SIZE);
+   }
+
+   @Test
+   public void testAggregationNotIndexed() {
+      CompletionStage<RestResponse> result = queryWithoutPagination(nonIndexedCache(), "SELECT count(field1), max(field2) FROM NotIndexedEntity");
+      final Long field1Count = getFieldAggregationValue(result, "field1");
+      assertEquals(NOT_INDEXED_ENTRIES, field1Count);
+   }
+
+   @Test
+   public void testAggregationNotIndexedPagination() {
+      CompletionStage<RestResponse> result = queryWithDefaultPagination(nonIndexedCache(), "SELECT count(field1), max(field2) FROM NotIndexedEntity");
+      final Long field1Count = getFieldAggregationValue(result, "field1");
+      assertEquals(NOT_INDEXED_ENTRIES, field1Count);
+   }
+
+   @Test
+   public void testCountOnlyNotIndexed() {
+      CompletionStage<RestResponse> result = queryWithPagination(nonIndexedCache(), "SELECT field1 FROM NotIndexedEntity", 0, 0);
+      assertTotalAndPageSize(result, NOT_INDEXED_ENTRIES, 0L);
+   }
+
+   private void assertTotalAndPageSize(CompletionStage<RestResponse> response, Long totalResults, Long pageSize) {
+      RestResponse restResponse = await(response);
+      String body = restResponse.getBody();
+      Json responseDoc = Json.read(body);
+      Json total = responseDoc.at("total_results");
+      if (totalResults == null) {
```

Comment:
could this be 0 instead of null ?

Reply:
I can't be null in the test, will fix it